metadata-ingestion/src/datahub/ingestion/source/tableau.py

```diff
@@ -273,6 +273,17 @@ def _create_upstream_table_lineage(
                 table.get("columns", []),
                 table_path,
             )
+
+        for datasource in datasource.get("upstreamDatasources", []):
+            datasource_urn = builder.make_dataset_urn(
+                self.platform, datasource.get("id", ""), self.config.env
```

Comment:
Hi @mayurinehate, should the name param here be `datasource["name"] + datasource.get("id", "")`?  Is the `id` optional? Also, do we need any updates to the documentation?

Reply:
1. Only dataset["id"] should be used in urn. It is consistent with other places where we generate urn in order to set lineage correctly.
2. Id is not optional. updated code for this commit
3. Added Lineage section in tableau doc to mention lineages emitted